### PR TITLE
Support get/remove/set retention policy command for topic

### PR DESCRIPTION
**Pull request by nodece**
Signed-off-by: Zixuan Liu <nodeces@gmail.com>

background from #246,  the PR implements the following commands:

- `pulsar topics get-retention <topic> -a` - Get the retention policy for a topic
- `pulsar topics remove-retention <topic>` - Remove the retention policy for a topic
- `pulsar topics set-retention <topic> --time <string> --size <string>` - Set the retention policy for a topic

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]

**Review by zymap — CHANGES_REQUESTED** [review]
it's better to add a test to verify the whole process. 
For example: set the retention to a topic and then get the value to verify is it set right. It can test both the cli and broker.

**Comment by nodece** [comment]
Hello @zymap, ut-tests always fail on Github Actions, but it is pass locally, could you give me some point?

File: pkg/ctl/topic/set_retention.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		RetentionTimeInMinutes: duration,
+	})
+	if err == nil {
+		vc.Command.Printf("Set the retention policy successfully on [%s]\n", topic)
+	}
+
+	return err
+}
+
+func parseRetentionSizeString(size string) (int64, error) {
```
Comment (zymap) [inline pkg/ctl/topic/set_retention.go:129]: I think we already have this method in utils package, why not use that?

File: pkg/ctl/topic/set_retention.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	}
+}
+
+var timeMinute = time.Minute.Seconds()
+var timeHour = time.Hour.Seconds()
+var timeDay = 24 * time.Hour.Seconds()
+var timeWeek = 7 * 24 * time.Hour.Seconds()
+var timeYear = 365 * timeDay
+
+func parseRetentionTimeString(duration string) (int64, error) {
```
Comment (zymap) [inline pkg/ctl/topic/set_retention.go:165]: This should already have in util package as well.

File: pkg/ctl/topic/set_retention.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+func doSetRetention(vc *cmdutils.VerbCmd, timeStr string, sizeStr string) error {
+	topic := vc.NameArg
+
+	if timeStr == "" {
+		return errors.New("missing option: --time")
+	}
+	if sizeStr == "" {
+		return errors.New("missing option: --size")
+	}
```
Comment (zymap) [inline pkg/ctl/topic/set_retention.go:100]: We can mark the flag as required

File: pkg/ctl/topic/retention_test.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/streamnative/pulsarctl/pkg/pulsar/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRetentionCmd(t *testing.T) {
+	topic := fmt.Sprintf("test-retention-topic-%d", time.Now().Unix())
```
Comment (zymap) [inline pkg/ctl/topic/retention_test.go:31]: we have a method `test.RandomSuffix()` to generate a random name

File: test/standalone.conf [hunk 5]
```diff
@@ -271,10 +271,17 @@ replicationProducerQueueSize=1000
 defaultRetentionTimeInMinutes=0
 
 # Default retention size
-defaultRetentionSizeInMB=0
+defaultRetentionSizeInMB = 0
```
Comment (zymap) [inline test/standalone.conf:274]: We don't use this configuration file to configure the test broker. We always use the environment to change the configurations. So you can remove this change.

**Comment by zymap** [comment]
@nodece I run the test on local as well. It actually failed. You can run a standalone and then run the test to check it.

File: pkg/ctl/topic/set_retention.go (hunk 1 above)
Comment (nodece) [reply pkg/ctl/topic/set_retention.go:129]: I don't found this method, could you point this method?

File: pkg/ctl/topic/set_retention.go (hunk 2 above)
Comment (nodece) [reply pkg/ctl/topic/set_retention.go:165]: I don't found this method, could you point this method?

File: pkg/ctl/topic/set_retention.go (hunk 3 above)
Comment (nodece) [reply pkg/ctl/topic/set_retention.go:100]: Could you point how to do?

Comment (zymap) [reply pkg/ctl/topic/set_retention.go:95]: https://github.com/streamnative/pulsarctl/blob/96382f197af9ef45cc37cea28320f8df8c99bd17/pkg/ctl/topic/revoke_permissions.go#L74

**Comment by flowchartsman** [comment]
@nodece I issued a [PR](https://github.com/nodece/pulsarctl/pull/1) against your PR branch to unify the tests and make them a bit like the dispatch tests that @limingnihao  did (which I in turn based my publish-rate command and tests on). Hope it helps!

**Review by zymap — APPROVED** [review]

**Pull request merged**